### PR TITLE
5 split feature tracker package ros and non ros pure c++

### PR DESCRIPTION
**Pull request by TanPinDa**

### CODE REVIEW

File: feature_tracker/CMakeLists.txt [hunk 1]
```diff
@@ -1,37 +1,62 @@
...
-
-find_package(catkin REQUIRED COMPONENTS
-    roscpp
-    std_msgs
-    sensor_msgs
-    cv_bridge
-    )
+cmake_minimum_required(VERSION 3.0)
+project(feature_tracker VERSION 1.0)
+set(CMAKE_CXX_FLAGS "-std=c++17 -DEIGEN_DONT_PARALLELIZE")
```
Comment (KeeJin) [inline feature_tracker/CMakeLists.txt:3]: Unsure about the necessity of the DEIGEN_DONT_PARALLELIZE flag

File: feature_tracker/CMakeLists.txt [hunk 2]
```diff
@@ -1,37 +1,62 @@
...
 
 find_package(OpenCV REQUIRED)
 find_package(camodocal REQUIRED)
+find_package(Eigen3 REQUIRED)
+find_package(spdlog REQUIRED)
```
Comment (KeeJin) [inline feature_tracker/CMakeLists.txt:8]: Need to update install-deps.sh for this new dependency

**Review by KeeJin — CHANGES_REQUESTED** [review]

File: feature_tracker/CMakeLists.txt (hunk 1 above)
Comment (TanPinDa) [reply feature_tracker/CMakeLists.txt:3]: Will remove. Artefact from copying. Guilty.

**Review by KeeJin — APPROVED** [review]
LGTM

**Pull request merged**